qa/1886

```diff
@@ -243,6 +243,30 @@ echo Compared to stdev:
 pmseries $args 'stdev_inst(kernel.all.uptime[count:10])'
 pmseries $args 'stdev_sample(kernel.all.uptime[count:10])'
 
+echo;echo "== Verify nth_percentile() functions for a non-singular metric"
+echo Original with no function:
+pmseries $args 'kernel.all.load[count:5]'
+echo Compared to nth_percentile_inst:
+pmseries $args 'nth_percentile_inst(kernel.all.load[count:5], 50)'
+echo Compared to nth_percentile_sample:
+pmseries $args 'nth_percentile_sample(kernel.all.load[count:5], 50)'
+
+echo;echo "== Verify nth_percentile() functions for a non-singular metric, only one instance"
+echo Original with no function:
+pmseries $args 'kernel.all.load{instance.name == "5 minute"}[count:5]'
+echo Compared to nth_percentile_inst:
+pmseries $args 'nth_percentile_inst(kernel.all.load{instance.name == "5 minute"}[count:5], 50)'
+echo Compared to nth_percentile_sample:
+pmseries $args 'nth_percentile_sample(kernel.all.load{instance.name == "5 minute"}[count:5], 50)'
+
+echo;echo "== Verify nth_percentile() functions for a singular metric"
+echo Original with no function:
+pmseries $args 'kernel.all.uptime[count:10]'
+echo Compared to nth_percentile_inst:
+pmseries $args 'nth_percentile_inst(kernel.all.uptime[count:10], 50)'
+echo Compared to nth_percentile_sample:
+pmseries $args 'nth_percentile_sample(kernel.all.uptime[count:10], 50)'
+
```

Comment:
@ShiyaoC this test could be improved by also checking error conditions in the input statements - i.e. making sure that we handle bad input for the 'nth_percentile_sample' expression - e.g. lacking parameters, has too many parameters, non-integer second parameter, too-large second parameter, etc.  The error handling should always produce consistent output, so its a good thing to add to the test to make sure we always handle it.

Reply:
I will add more tests for edge cases.